tests: flaky TestConnectionHandlerOpenUpdateClose

```
2025-11-20T15:44:59.9010260Z === RUN   TestConnectionHandlerOpenUpdateClose
2025-11-20T15:44:59.9208910Z 2025/11/20 15:44:59 tarantool: connect to 127.0.0.1:3013 failed: failed to identify: read error: context canceled
2025-11-20T15:44:59.9312900Z     connection_pool_test.go:1138: 
2025-11-20T15:44:59.9564500Z         	Error Trace:	/Users/runner/work/go-tarantool/go-tarantool/tarantool/go-tarantool/pool/connection_pool_test.go:1138
2025-11-20T15:44:59.9567600Z         	Error:      	Expected nil, but got: &errors.errorString{s:"can't find rw instance in pool"}
2025-11-20T15:44:59.9676660Z         	Test:       	TestConnectionHandlerOpenUpdateClose
2025-11-20T15:44:59.9777130Z         	Messages:   	failed to make ro
2025-11-20T15:44:59.9879310Z --- FAIL: TestConnectionHandlerOpenUpdateClose (0.02s)
2025-11-20T15:44:59.9982560Z === RUN   TestConnectionHandlerOpenError
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -33,6 +33,9 @@ Versioning](http://semver.org/spec/v2.0.0.html) except to the first release.
 
 ### Fixed
 
+* Fixed the fluctuating behavior of the TestConnectionHandlerOpenUpdateClose
+  test by increasing the waiting time (#502).
+
 ## [v2.4.1] - 2025-10-16
 
 This maintenance release marks the end of active development on the `v2`

--- a/pool/connection_pool_test.go
+++ b/pool/connection_pool_test.go
@@ -1123,7 +1123,7 @@ func TestConnectionHandlerOpenUpdateClose(t *testing.T) {
 
 	h := &testHandler{}
 	poolOpts := pool.Opts{
-		CheckTimeout:      100 * time.Microsecond,
+		CheckTimeout:      100 * time.Millisecond,
 		ConnectionHandler: h,
 	}
 	connPool, err := pool.ConnectWithOpts(ctx, poolInstances, poolOpts)

